normandy/control/static/control/js/utils/formHelpers.js

```diff
@@ -0,0 +1,23 @@
+/* Redux forms with complex/nested data structures require
+ * a specialized fields array to handle objects and arrays.
+ * Eg: ['surveyId', 'surveys[].title', 'defaults.thankYouMessage']
+ */
+const reduxFormFields = {
```

Comment:
That makes things a little tricky, because we actually need the form fields ready and available in RecipeForm.jsx. They have to be passed into the ActionForm component as soon as it initializes. Dynamically adding and changing redux form fields is not really possible without completely destroying and re-initializing the form.